### ISSUE TITLE
After Cast Delay Reduction Algorithm

### ISSUE THREAD

**Issue by Everade** [rathena/rathena#6135]
* **rAthena Hash**: 
3115a1d39ce16428b74cef681089fb39a6bcfada

* **Client Date**: 
2021-01-07aRagexeRE

* **Server Mode**: 
Renewal

* **Description of Issue**: 
  * Result: 
  The After Cast Delay Algorithm doesn't match.
  The skill Poem of Bragi (Magic Strings) after cast delay reduction seems to be multiplicative instead of a total sum of all reductions combined.
  Testing with Kiel-D-01 + Poem of Bragi delivers only a x 0.49 reduction instead of x 0.6.
  The issue could possibly be caused by being calculated similiarly to the Variable Cast time:
  `a x ( 1 - b ÷ 100 ) x ( 1 - c ÷ 100 )  = 1470ms`
  a = Tarot Card of Fate after cast delay (3000ms)
  b = Poem of Bragi reduction (30)
  c = Kiel-D-01 reduction (30)

  * Expected Result:
  `Cast Delay ( seconds ) = BaseDelay × ( 1 − Sum_DelReduc ÷ 100 )`
  BaseDelay = Base Cast Delay of a skill.
  Sum_DelReduc = Sum of reductions for Cast Delay.

  Which would result in the expected reduction of:
  `a x (( 1 - b ÷ 100 ) + ( 1 - c ÷ 100 )) = 1200ms`
  a = Tarot Card of Fate after cast delay (3000ms)
  b = Poem of Bragi reduction (30)
  c = Kiel-D-01 reduction (30)
  In short: calculates first skill based reduction, then gear. Instead of a deduction of the entire sum.

  * How to Reproduce: 
  Use both Poem of Bragi (skill) and a Kiel-D-01 Card.
  Then check the after-cast timer of Tarot Card of Fate. (original 3000ms)
  The result will be 1470ms instead of  the expected 1200ms.
 

  * Official Information: 
https://irowiki.org/wiki/Skills#Cast_Delay
Unfortunately i don't have any KRO source.

* **Modifications that may affect results**: 
  None

* **Additional Information**: 
  I haven't tested any other skills. It's possibly an isolated case.

**Comment by Everade** [comment]
I use a custom command to see the actual miliseconds. However since that might not be believable, i made an obvious setup that the visual cooldown indicator on the skill icons make are easy to see. (just count the seconds)

**Setup as follows:**
Tarot Card of Fate with a custom After Cast Delay of 10seconds.
Poem of Bragi is official, gives a -30% reduction on lvl 10.
Kiel Card with a custom value of -70% reduction.
So the result would be the sum of -100% after cast delay reduction.

The results however show that there's still a roughly 2seconds after cast delay, **as the official skill icon cooldown indicators in the top left show.**

I hope this proofs my claims.

https://user-images.githubusercontent.com/12261601/130696197-f0d0b383-a5dd-4211-bf99-af937a095f01.mp4
